tests/test_data.py

```diff
@@ -284,3 +288,72 @@ def load_prediction_data(self, *args, **kwargs):
             DatasetError, match="An empty dataset was returned by load_prediction_data"
         ):
             regression.make_prediction(data, 0)
+
+
+class TestDemoDatasetModule:
+    @pytest.fixture
+    def load_dataset_iris(self) -> Dataset:
+        return load_demo_dataset("iris")
+
+    def test_repr_is_correct_load(self, load_dataset_iris: Dataset):
+        result = str(load_dataset_iris)
+        assert result == "<DemoData - Dataset>"
+
+    def test_dataset_x_attribute_access_works_correctly(
+        self, load_dataset_iris: Dataset
+    ):
+        assert load_dataset_iris._x is None
+        features = load_dataset_iris.x
+        assert load_dataset_iris._x is not None
+        assert len(features) == 150
+
+    def test_dataset_y_attribute_access_works_correctly(
+        self, load_dataset_iris: Dataset
+    ):
+        assert load_dataset_iris._y is None
+        features = load_dataset_iris.y
+        assert load_dataset_iris._y is not None
+        assert len(features) == 150
```

Comment:
We don't need to test that the dataset class works - that is tested somewhere else. I would suggest testing that the Iris data comes out correctly - matching the sklearn data